bin/check-ssl-root-issuer.rb

```diff
@@ -0,0 +1,127 @@
+#! /usr/bin/env ruby
+#
+#   check-ssl-root-issuer
+#
+# DESCRIPTION:
+#   Check that a certificate is chained to a specific root certificate issuer
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# USAGE:
+#
+#   Check that a specific website is chained to a specific root certificate
+#       ./check-ssl-root-issuer.rb \
+#           -u https://example.com \
+#           -i "CN=DST Root CA X3,O=Digital Signature Trust Co."
+#
+# LICENSE:
+#   Copyright Jef Spaleta (jspaleta@gmail.com) 2020
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'openssl'
+require 'uri'
+require 'net/http'
+require 'net/https'
+
+#
+# Check root certificate has specified issuer name
+#
+class CheckSSLRootIssuer < Sensu::Plugin::Check::CLI
+  option :url,
+         description: 'Url to check: Ex "https://google.com"',
+         short: '-u',
+         long: '--url URL',
+         required: true
+
+  option :issuer,
+         description: 'An X509 certificate issuer name, RFC2253 format Ex: "CN=DST Root CA X3,O=Digital Signature Trust Co."',
+         short: '-i',
+         long: '--issuer ISSUER_NAME',
+         required: true
+
+  option :regexp,
+         description: 'Treat the issuer name as a regexp',
+         short: '-r',
+         long: '--regexp',
+         default: false,
+         boolean: true,
+         required: false
+
+  option :format,
+         description: 'optional issuer name format. Defaults to RFC2253. Allowed values: RFC2253, ONELINE, COMPAT',
+         short: '-f',
+         long: '--format FORMAT_VAL',
+         default: 'RFC2253',
+         required: false
+
+  def validate_opts
```

Comment:
A couple thoughts:
- I think we should we still add validation to the options passed in themselves. if I pass in an invalid value I see no reason to run this code here
- I think we should add conditional logic, if I specify something other than those two options the way the code is written it will attempt to validate with `RFC2253` regardless of what is passed in
- should we add a qualifier here that this is specific to format or do you intend that we would keep adding more validation to this?


Maybe that should be something we pass in and can try something like this? (not sure if this is valid)
```ruby
def validate_format_opts(config[:format])
  OpenSSL::X509::Name::config[:format]
end
```
if we cant specify it like that then we could always use if, elif, else or case statements to handle.

Reply:
Okay I've updated the logic to use mixin's cli option  value validation method.